UnityProject/Assets/Tests/CraftingTests.cs

```diff
@@ -85,5 +89,137 @@ public void CheckCraftingListTest()
 			Logger.Log(report.ToString(), Category.Tests);
 			Assert.IsEmpty(report.ToString());
 		}
+
+		/// <summary>
+		/// 	Recipes have links to their ingredients, and the ingredients INCLUDING their heirs should have a link
+		/// 	to the recipe. This test is supposed to find all ingredients that miss their link to the recipe
+		/// 	or have a wrong recipe index.
+		/// </summary>
+		[Test]
+		public void CheckIngredientCrossLinks()
+		{
+			StringBuilder report = new StringBuilder();
+			// thank unity we have no better way to see a base (parent) or variants (heirs) of game objects, so
+			// we have to search it using .prefab files. This dictionary contains pairs of values:
+			// <Parent, List<parent's heirs>> or something like that. Heirs can also have its heirs, so they also
+			// can be parents and should be presented in this dictionary with the matching key.
+			Dictionary<GameObject, HashSet<GameObject>> parentsAndChilds =
+				new Dictionary<GameObject, HashSet<GameObject>>();
+			string[] possibleIngredientPrefabGuids =
+				AssetDatabase.FindAssets("t:prefab", new string[] {"Assets/Prefabs"});
+
+			foreach (string possibleIngredientPrefabGuid in possibleIngredientPrefabGuids)
+			{
+				GameObject possibleChildIngredient = AssetDatabase.LoadAssetAtPath<GameObject>(
+					AssetDatabase.GUIDToAssetPath(possibleIngredientPrefabGuid)
+				);
+				// i don't know how but maybe someone sometime will just remove this component from a game object...
+				if (possibleChildIngredient.GetComponent<CraftingIngredient>() == null)
+				{
+					continue;
+				}
+				// assuming that game object as a prefab variant and trying to find its base
+				GameObject parent = PrefabExtensions.GetVariantBaseGameObject(possibleChildIngredient);
+				// is it not a variant?
+				if (parent == null)
+				{
+					// yes so no need to add it somewhere
+					continue;
+				}
+
+				// have we met this ingredient first time?
+				if (parentsAndChilds.ContainsKey(parent) == false)
+				{
+					parentsAndChilds[parent] = new HashSet<GameObject>();
+					continue;
+				}
+
+				parentsAndChilds[parent].Add(possibleChildIngredient);
+			}
+
+			// yes we can search without this recipesPath but i don't wanna make this test run for 5 minutes
+			string[] recipeGuids = AssetDatabase.FindAssets("t:ScriptableObject", new string[] {recipesPath});
```

Comment:
```suggestion
			string[] recipeGuids = AssetDatabase.FindAssets("t:CraftingRecipe", new string[] {recipesPath});
```

Pretty sure you can do that ^ instead